packages/assembly/index.ts

```diff
@@ -228,7 +228,7 @@ export namespace ASON {
       let entry = customTable.allocate();
       let entryId = this.entryId++;
       entry.entryId = entryId;
-      entry.rtId = idof<U>();
+      entry.rtId = getObjectType(changetype<usize>(value));
```

Comment:
Instead of calling `idof<U>()` we actually inspect the rttid of the reference here.